beacon_node/beacon_chain/src/execution_proof_verification.rs

```diff
@@ -179,7 +178,9 @@ fn validate_execution_proof_for_gossip<T: BeaconChainTypes, O: ObservationStrate
     }
 
     // TODO: Add timing validation based on slot
-    // TODO: Add duplicate proof detection
+    if O::observe() {
+        // TODO: Add duplicate proof detection
```

Comment:
Related to this, one assumption also being made is that the block will arrive before the proofs. This assumption is reasonably sound given the time it takes to create proofs, however I can see it being broken in the following cases:

- The slot time is decreased (implying the execution payload gas limit is lowered) which would mean that proofs are created faster
- A node is not well-connected and does not receive the block until after the proof due to network lag. This may happen more frequently for empty blocks.

One solution here would be to store proofs for blocks currentSlot + N in the future, but we would need to store the slot number also in the execution proof message because an unknown beacon root cannot tell you how far in the future it is